src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,110 @@
 package com.es.phoneshop.model.product;
 
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
```

Comment:
class name is not good enough
it should not say about the resources which are used: if you decide to change the ArrayList to any other collection you will need to rename the class name, class name says about the business logic. For example CustomProductDao, or ProductDaoImpl (because its an implementation of interface ProductDao)
Please read about namings